coordinator/src/db/funding_rates.rs

```diff
@@ -0,0 +1,89 @@
+use crate::schema::funding_rates;
+use anyhow::Context;
+use anyhow::Result;
+use diesel::prelude::*;
+use rust_decimal::prelude::FromPrimitive;
+use rust_decimal::prelude::ToPrimitive;
+use rust_decimal::Decimal;
+use time::OffsetDateTime;
+use xxi_node::commons::to_nearest_hour_in_the_past;
+
+#[derive(Queryable, Debug)]
+struct FundingRate {
+    #[diesel(column_name = "id")]
+    _id: i32,
+    start_date: OffsetDateTime,
+    end_date: OffsetDateTime,
+    rate: f32,
+    #[diesel(column_name = "timestamp")]
+    _timestamp: OffsetDateTime,
+}
+
+pub(crate) fn insert(
+    conn: &mut PgConnection,
+    funding_rates: &[xxi_node::commons::FundingRate],
+) -> Result<()> {
+    conn.transaction(|conn| {
+        for funding_rate in funding_rates {
+            insert_one(conn, funding_rate)
+                .with_context(|| format!("Failed to insert funding rate: {funding_rate:?}"))?;
+        }
+
+        Ok(())
+    })
+}
+
+fn insert_one(conn: &mut PgConnection, params: &xxi_node::commons::FundingRate) -> QueryResult<()> {
+    let affected_rows = diesel::insert_into(funding_rates::table)
+        .values(&(
+            funding_rates::start_date.eq(params.start_date()),
+            funding_rates::end_date.eq(params.end_date()),
+            funding_rates::rate.eq(params.rate().to_f32().expect("to fit")),
+        ))
+        .execute(conn)?;
+
+    if affected_rows == 0 {
+        return Err(diesel::result::Error::NotFound);
+    }
+
+    Ok(())
+}
+
+// TODO: Check that `end_date` isn't already in the past.
+pub(crate) fn get_next_funding_rate(
+    conn: &mut PgConnection,
+) -> QueryResult<Option<xxi_node::commons::FundingRate>> {
+    let funding_rate: Option<FundingRate> = funding_rates::table
+        .order(funding_rates::end_date.desc())
+        .first::<FundingRate>(conn)
+        .optional()?;
+
+    let funding_rate = funding_rate.map(xxi_node::commons::FundingRate::from);
+
+    Ok(funding_rate)
+}
+
+/// Get the funding rate with an end date that is equal to the current date to the nearest hour.
+pub(crate) fn get_funding_rate_charged_in_the_last_hour(
+    conn: &mut PgConnection,
+) -> QueryResult<Option<xxi_node::commons::FundingRate>> {
+    let now = OffsetDateTime::now_utc();
+    let now = to_nearest_hour_in_the_past(now);
+
+    let funding_rate: Option<FundingRate> = funding_rates::table
```

Comment:
I don't think this works because the table might contain the _next_ funding rate as well, e.g. the db might have
1.  from 12:00-13:00, 5%
2. from: 10:00-11:59 3%

and we are interested the 2. when we call it at e.g. 11:21.